composer.json

```diff
@@ -11,16 +11,17 @@
         }
     ],
     "require": {
-        "php": "^5.4|7.*",
-        "puli/composer-plugin": "1.0.0-beta9"
+        "php": "^5.4|7.*"
     },
     "require-dev": {
         "php-http/httplug": "^1.0",
         "php-http/message-factory": "^1.0",
+        "puli/composer-plugin": "1.0.0-beta9",
         "phpspec/phpspec": "^2.4",
         "henrikbjorn/phpspec-code-coverage" : "^1.0"
     },
     "suggest": {
+        "puli/composer-plugin": "Allow to use Discovery",
```

Comment:
You would include discovery to ensure that the classes provided by discovery are available. Although discovery will throw an exception if puli is not available. The idea is that Puli composer plugin is an environmental detail: you can install it in your application, if you want to use it. As discovery should always be a fallback, the fact that you have it installed does not mean that you actually use the discovery layer. This makes discovery optional (although our wrapper around Puli must be a hard dependency wherever it is used).